### ISSUE TITLE
Store recordings on S3

### ISSUE THREAD

**Issue by ffdixon** [bigbluebutton/bigbluebutton#3890]
Extend the record and playback scripts to give the administrator the option of configuring a BigBlueButton server to store the recordings on Amazon S3.

This would essentially give unlimited storage of recordings to a BigBlueButton server.

**Comment by kirkjaa** [comment]
Yes this feature would help greatly with production server where it has been deployed for an extended period and need to keep archive of the recordings.

**Comment by APUTSA** [comment]
@ffdixon My website uses BBB for live tutoring and live courses. I am currently shifting it to amazon EC2. Do you recommend S3, EBS or EFS to store the recordings?

**Comment by ffdixon** [comment]
We recommend using EBS to store the recordings (you could map the /var/bigbluebutton directory to the EBS mount).  

This way, you could ideally mount the EBS volume on a new server as a way to "transfer" over your recordings to the new server.

**Comment by stsecole** [comment]
> 
> 
> @ffdixon My website uses BBB for live tutoring and live courses. I am currently shifting it to amazon EC2. Do you recommend S3, EBS or EFS to store the recordings?

I need your help on how I can setup BBB on a shared host ?

**Comment by muzammilathena** [comment]
Hi ffdixon,

I am having a online tutoring website i want to map my publish and raw dir to s3 how can i achieve it.

**Comment by rohitcoder** [comment]
You can use this to store files on s3 https://github.com/rohitcoder/BBB-AWS-S3

**Comment by GhaziTriki** [comment]
@ffdixon I would name it `Store recordings on S3 compatible storage` as some providers do not explicitly use S3 but they make they object storage S3 compatible.

**Comment by vahid75** [comment]
Hi there . anyone knows how is this feature going? is this implement now or not? i checked at official documentation but noting related to s3 was found

**Comment by hostbbb** [comment]
This is doable today with some customizations, just not a standard config option in BBB.   Couple approaches.

1) mount as s3 fuse directory ( i dont use, since losing this mount will cause loss of recording)
2) just push the objects as part of post_publish.rb and rewrite the playback url

the issue is metadata.xml needs to be read/writable as the state of recording is maintained in this file, and the recording api calls need to edit this.

Also some moodle plugins will not work out of the box if everything is on s3, require slight changes

**Comment by mabras** [comment]
ping
